mathesar/imports/excel.py

```diff
@@ -29,28 +23,23 @@ def insert_data_from_excel_data_file(name, schema, column_names, engine, comment
     insert_records_from_excel(
         table,
         engine,
-        excel_filepath,
+        dataframe,
     )
     return table
 
 
 def create_db_table_from_excel_data_file(data_file, name, schema, comment=None):
     db_name = schema.database.name
     engine = create_mathesar_engine(db_name)
-    excel_filepath = data_file.file.path
-    column_names = process_column_names(
-        get_column_names_from_excel(excel_filepath)
-    )
+    dataframe = pandas.read_excel(data_file.file.path)
+    column_names = process_column_names(dataframe.columns)
     try:
-        table = insert_data_from_excel_data_file(name, schema, column_names, engine, comment, excel_filepath)
+        table = insert_data_from_excel_data_file(name, schema, column_names, engine, comment, dataframe)
```

Comment:
this should be renamed to 

```suggestion
        table = insert_data_from_dataframe(name, schema, column_names, engine, comment, dataframe)
```
because, you're inserting from a dataframe here, it coming from excel is irrelevant (correct me if I'm wrong), or even

```suggestion
        table = insert_records_from_dataframe(name, schema, column_names, engine, comment, dataframe)
```

because, you refered to data as `records` in another place: both `data` and `records` work, imo, but you want to be consistent with how you're referring to rows/data/records.